### DISCUSSION TITLE
Text with ANSI escape codes provokes glitches in a Log widget

### DISCUSSION THREAD

**Discussion by superlevure** [Textualize/textual#3507] [General]
- [x] Have you checked closed issues? https://github.com/Textualize/textual/issues?q=is%3Aissue+is%3Aclosed

When writing text that contains ANSI escape codes to a `Log` widget, the border glitches: 

<img width="1334" alt="image" src="https://github.com/Textualize/textual/assets/26854828/a5735ec6-21b3-4ff5-9624-f508954a4fbc">

I've posted a minimal reproducible example here: https://github.com/superlevure/textual-scroll-bug

For context, the data I'm trying to write to a `Log` widget comes from a Gitlab CI job trace, as produced by this API: https://docs.gitlab.com/ee/api/jobs.html#get-a-log-file

# Textual Diagnostics

## Versions

| Name    | Value  |
|---------|--------|
| Textual | 0.39.0 |
| Rich    | 13.6.0 |

## Python

| Name           | Value                                                                       |
|----------------|-----------------------------------------------------------------------------|
| Version        | 3.11.5                                                                      |
| Implementation | CPython                                                                     |
| Compiler       | GCC 11.4.0                                                                  |
| Executable     | .env/bin/python3.11 |

## Operating System

| Name    | Value                                                |
|---------|------------------------------------------------------|
| System  | Linux                                                |
| Release | 5.4.0-1106-gcp                                       |
| Version | #115~18.04.1-Ubuntu SMP Mon May 22 20:46:39 UTC 2023 |

## Terminal

| Name                 | Value           |
|----------------------|-----------------|
| Terminal Application | vscode (1.83.0) |
| TERM                 | xterm-256color  |
| COLORTERM            | truecolor       |
| FORCE_COLOR          | *Not set*       |
| NO_COLOR             | *Not set*       |

## Rich Console options

| Name           | Value                |
|----------------|----------------------|
| size           | width=187, height=33 |
| legacy_windows | False                |
| min_width      | 1                    |
| max_width      | 187                  |
| is_terminal    | True                 |
| encoding       | utf-8                |
| max_height     | 33                   |
| justify        | None                 |
| overflow       | None                 |
| no_wrap        | False                |
| highlight      | None                 |
| markup         | None                 |
| height         | None                 |

**Comment by willmcgugan** [comment]
As a rule, you can't use raw escape sequences in Textual. But you can convert these to something Textual can work with by calling `Text.from_ansi`. You can import the Text class from rich.text

IIRC the `Log` widget only works with plain text, but you can use the `RichLog` widget which supports rich text.

**Reply by superlevure** [reply]
And it works perfectly, thank you @willmcgugan! And sorry I missed that in the documentation
